internal/helm-operator-plugins/predicate/depedent_test.go

```diff
@@ -0,0 +1,132 @@
+package predicate
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+func TestDependentPredicateFuncsCreate(t *testing.T) {
+	for _, tt := range []struct {
+		description string
+		arg         event.CreateEvent
+		result      bool
+	}{
+		{
+			description: "Happy path",
+			arg: event.CreateEvent{
+				Object: &unstructured.Unstructured{
+					Object: map[string]interface{}{"key": "Value"},
+				},
+			},
+			result: false,
+		},
+	} {
+		t.Run(tt.description, func(t *testing.T) {
+			funcs := DependentPredicateFuncs()
+			result := funcs.CreateFunc(tt.arg)
+			require.Equal(t, tt.result, result)
+		})
+	}
+}
+
+func TestDependentPredicateFuncsDelete(t *testing.T) {
+	for _, tt := range []struct {
+		description string
+		arg         event.DeleteEvent
+		result      bool
+	}{
+		{
+			description: "Happy path",
+			arg: event.DeleteEvent{
+				Object: &unstructured.Unstructured{
+					Object: map[string]interface{}{"key": "Value"},
+				},
+			},
+			result: true,
+		},
+	} {
+		t.Run(tt.description, func(t *testing.T) {
+			funcs := DependentPredicateFuncs()
+			result := funcs.DeleteFunc(tt.arg)
+			require.Equal(t, tt.result, result)
+		})
+	}
+}
+
+func TestDependentPredicateFuncsGeneric(t *testing.T) {
+	for _, tt := range []struct {
+		description string
+		arg         event.GenericEvent
+		result      bool
+	}{
+		{
+			description: "Happy path",
```

Comment:
nit: for these descriptions, could we update them to explain what the happy path is supposed to be?

Reply:
Thanks!  I add some descriptions.